pyccl/halos/profiles_cib.py

```diff
@@ -189,21 +189,19 @@ def _Lumcen(self, M, a):
         return Lumcen
 
     def _Lumsat(self, M, a):
-        Lumsat = np.zeros_like(M)
-        # Loop over Mparent
-        # TODO: if this is too slow we could move it to C
-        # and parallelize
-        for iM, Mparent in enumerate(M):
-            if Mparent > self.Mmin:
-                # Array of Msubs (log-spaced with 10 samples per dex)
-                nm = max(2, int(np.log10(Mparent/1E10)*10))
-                msub = np.geomspace(1E10, Mparent, nm+1)
-                # Sample integrand
-                dnsubdlnm = self.dNsub_dlnM_TinkerWetzel10(msub, Mparent)
-                Lum = self._Lum(np.log10(msub), a)
-                integ = dnsubdlnm*Lum
-                Lumsat[iM] = simps(integ, x=np.log(msub))
-        return Lumsat
+        res = np.zeros_like(M)
+        M_use = M[M >= self.Mmin, None]
+        logM = np.log10(M_use)
+        LOGM_MIN = 10
```

Comment:
It seems the model actually uses `Mmin` as the lower limit of this integral too (see Eq. 15 of https://arxiv.org/pdf/1109.1522.pdf). Can you make this modification?

Reply:
Which line does this refer to? I can't see where `Mmin` isn't used as the lower limit.

Reply:
self.Mmin

Reply:
oh okay, done